.circleci/config.yml

```diff
@@ -0,0 +1,214 @@
+      - run:
+          name: gojson
+          command: |
+            if ! [ -x $GOPATH/bin/gojson ] ; then
+              go get -u github.com/ChimeraCoder/gojson/gojson
+            fi
+      - run:
+          name: dep
+          command: |
+            if ! [ -x $GOPATH/bin/dep ] ; then
+              curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            fi
+      - run:
+          name: stringer
+          command: |
+            if ! [ -x $GOPATH/bin/stringer ] ; then
+              go get -u -u golang.org/x/tools/cmd/stringer
+              cd $GOPATH/src/golang.org/x/tools/cmd/stringer
+              git reset --hard <<parameters.stringer_rev>>
+              go install
+            fi
+      - save_cache:
+          key: "protoc-<<parameters.protoc_version>>_gogofaster-<<parameters.protoc_gen_gogofaster_tag>>_stringer-<<parameters.stringer_rev>>_<<parameters.cachebust>>"
+          paths:
+            - /opt/protoc
+            - /go/bin/protoc-gen-gogofaster
+            - /go/src/github.com/gogo/protobuf
+            - /go/bin/dep
+            - /go/bin/stringer
+  fetch_and_build_latest_go:
+    description: "Builds the latest version of go from source"
+    steps:
+      - run:
+          name: create datestamp
+          command: |
+            echo go-master-$(date +%Y-%m-%d) | tee /tmp/go-version
+      - run:
+          name: "Move away stable go"
+          command: |
+            if ! [ -d /usr/local/go-stable ] ; then
+              mv /usr/local/go /usr/local/go-stable
+            fi
+      - restore_cache:
+          key: |
+            {{ checksum "/tmp/go-version" }}
+      - run:
+          name: get latest master
+          command: |
+            if [ "$(cat /tmp/go-version)" = "$(cat /usr/local/go/VERSION)" ] ; then
+              exit 0
+            fi
+            wget -O/tmp/go-master.tgz https://github.com/golang/go/archive/master.tar.gz
+            tar -C /tmp -zxf /tmp/go-master.tgz
+            mv /tmp/go-master /usr/local/go
+            export GOSRC=/usr/local/go
+            export GOROOT=$GOSRC
+            export GOROOT_BOOTSTRAP=/usr/local/go-stable
+            export GOBUILD=$GOSRC/src
+            cp /tmp/go-version $GOROOT/VERSION
+            cd $GOBUILD
+            ./make.bash
+      - save_cache:
+          key: |
+            {{ checksum "/tmp/go-version" }}
+          paths:
+            - /usr/local/go
+  test:
+    description: "Run the tests"
+    steps:
+      - run:
+          name: go test
+          command: |
+            mkdir -p /tmp/test-results/go
+            trap "go get -u github.com/jstemmer/go-junit-report && go-junit-report </tmp/go-test-output.txt > /tmp/test-results/go/test-report.xml" EXIT
+            go test -race -v -timeout 60s ./... | tee /tmp/go-test-output.txt
+      - store_test_results:
+          path: /tmp/test-results/go
+
+jobs:
+  success:
+    docker:
+      - image: alpine:latest
+    steps:
+      - run: echo yay
```

Comment:
Does this do anything? (Why do we need to print something specific for success here - is it not marked otherwise in the build?)

Reply:
The `success` job doesn't do anything! But it does collect all the other jobs as dependencies. I think we can have all jobs be independent of each other (I cargo-culted that from another of my repos' CI configs) (: